api-docs: Update "API keys", use `start_tabs` formatting, and cross-link related articles.

drrosa opened the pull request:
- Updates API doc to follow current help center documentation patterns.
- Updates API docs that have numbered instructions to use the `{start_tabs}` formatting we use in the help center.
- Cross-links related articles.

Fixes #28876.

**Screenshots and screen captures:**
-  https://zulip.com/api/api-keys
![image](https://github.com/zulip/zulip/assets/2343554/e226c223-6948-4308-83a1-9c0488725690)

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

drrosa commented:
@alya This PR is ready for review.
@zulipbot add "documentation review"

alya commented:
Works for me! @laurynmm any feedback on this one?

alya commented:
I haven't reviewed the cross-links super carefully.

laurynmm commented on api_docs/deploying-bots.md:
```diff
@@ -36,6 +36,8 @@ Zulip Botserver starts a web server that listens to incoming messages
 from your main Zulip server. The sequence of events in a successful
 Botserver interaction are:
 
+{start_tabs}
```
This section isn't instructions, but rather describing a sequence of events. If we want to keep these tabs as instruction blocks, maybe we shouldn't add them here.

Also, noting that the code highlighting for the json code block in the below bullet points is broken.

drrosa replied:
Yeah, I agree. It's probably best to use tabs only for instruction blocks.

Yep, code highlighting is broken within a numbered list, and it looks like it can't be fixed just by tweaking the formatting so I removed it for now.

![image](https://github.com/zulip/zulip/assets/2343554/0063f609-e520-4691-90ad-d937ff01d704)

laurynmm replied:
Yeah, this looks better. Can we fix that final sentence and make it part of the 4th point?

laurynmm commented on api_docs/deploying-bots.md:
```diff
@@ -216,13 +229,17 @@ running it manually.
 
 ### Troubleshooting
 
+{start_tabs}
```
These are also not instructions, so maybe should be formatted in another way.

drrosa replied:
Hmm, maybe as bullet points instead?

![image](https://github.com/zulip/zulip/assets/2343554/17b019db-dc64-470d-b628-dc4c984569bd)

laurynmm replied:
That second bullet point looks odd. I think the second sentence should either be in the same paragraph or there should be a paragraph break.

laurynmm commented on api_docs/incoming-webhooks-overview.md:
```diff
@@ -197,3 +197,7 @@ Parameters accepted in the URL include:
+
+## Related articles
+
+* [Integrations overview](/api/integrations-overview)
```
Even though it's linked in the article, I'd link to the example incoming webhook walk through here as well. And maybe the non-webhook integrations article as well.

laurynmm commented on api_docs/incoming-webhooks-walkthrough.md:
```diff
@@ -475,20 +479,24 @@ When you have finished your webhook integration and are ready for it to be
 available in the Zulip product, follow these steps to prepare your pull
 request:
 
+{start_tabs}
```
This should also maybe be rewritten as tips / suggestions instead of a set of instructions.

drrosa replied:
How about:

![image](https://github.com/zulip/zulip/assets/2343554/4897c0db-53e1-44a5-8249-12049717b83f)

laurynmm replied:
This looks good to me.

laurynmm commented on api_docs/running-bots.md:
```diff
@@ -20,6 +20,8 @@ You'll need:
 
 **Note: Please be considerate when testing experimental bots on public servers such as chat.zulip.org.**
 
+{start_tabs}
```
This block of instructions could use a little clean-up while we're here. In particular, the last three steps all have bits that we could improve.

drrosa replied:
Yes, I agree. Hmm, I need to think a bit more about how to clean this one up.

drrosa replied:
![image](https://github.com/zulip/zulip/assets/2343554/fb3a5dfe-805a-43ca-bf68-f2563c237dee)

laurynmm commented on api_docs/writing-bots.md:
```diff
@@ -17,24 +17,28 @@ On this page you'll find:
 
 ## Installing a development version of the Zulip bots package
 
+{start_tabs}
```
Another spot where we could clean up the instructions. They are the reverse of what we normally do with the instruction text after the code / command line example.

drrosa replied:
How about:

![image](https://github.com/zulip/zulip/assets/2343554/ee4c2bf2-5fb1-41cc-a871-b64a9d9064cb)

laurynmm commented on api_docs/writing-bots.md:
```diff
@@ -17,24 +17,28 @@ On this page you'll find:
+1. *Finished*. You should now see the name of your venv preceding your prompt,
    e.g. `(zulip-api-py3-venv)`.
 
 *Hint: `provision` installs the `zulip`, `zulip_bots`, and
```
Shouldn't this be formatted in our tip markdown formatting? Instead of being in the instruction tabs block.

laurynmm reviewed (commented):
@drrosa - Thanks for doing these updates! I think there are a few places where we should revise the numbered bullet points, instead of just wrapping them in the start/end tabs. And a few places where we can update / clean up the existing text since we're looking at these numbered bullets.

Let me know if you have any questions about my comments below. I checked the related article links and those all seem to be good and working in these changes.

drrosa commented:
> @drrosa - Thanks for doing these updates! I think there are a few places where we should revise the numbered bullet points, instead of just wrapping them in the start/end tabs. And a few places where we can update / clean up the existing text since we're looking at these numbered bullets.

@laurynmm Thanks for the thorough review! I updated the PR in response to your feedback and added screenshots corresponding to your comments above.

timabbott commented:
This needs a rebase after https://github.com/zulip/zulip/pull/28952.

timabbott commented on api_docs/deploying-bots.md:
````diff
@@ -44,22 +44,21 @@ Botserver interaction are:
 1. The Zulip server sends a POST request to the Botserver on `https://bot-server.example.com/`:
 
-    ```json
+    ```
````
What's the thinking on removing the `json` language formatting? I can imagine a few reasons, but am not sure what the motivation was.

drrosa replied:
> What's the thinking on removing the `json` language formatting? I can imagine a few reasons, but am not sure what the motivation was.

Code highlighting was broken as noted here https://github.com/zulip/zulip/pull/28895#discussion_r1498550383.

> I'm also a bit confused by the whitespace changes just below.

Good catch. I think I unintentionally changed the whitespace formatting while trying to get the syntax highlighting to work. I reverted the changes.

timabbott replied:
OK -- can you start a thread in #issues about syntax highlighting being broken here? I doubt it's a hard fix, given that I think it does work in other places.

timabbott commented on api_docs/writing-bots.md:
```diff
@@ -17,23 +17,35 @@ On this page you'll find:
+   repository:
 
-3. `python3 ./tools/provision` - install all requirements in a Python virtualenv.
+        git clone https://github.com/zulip/python-zulip-api.git
```
In general we have a practice of using fenced code blocks in the API documentation; they're a lot easier to reason about their indentation levels in relation to bulleted lists.

timabbott commented:
This lgtm as an edit pass on this; I'm sure there's more we could do, and it wouldn't be a bad idea to spend more time on it, but I'm comfortable merging this after the last few questions are addressed and fixing forward anything further.

timabbott commented:
Merged, thanks @drrosa!
